### PR TITLE
Initial firestore mapping table implementation

### PR DESCRIPTION
**Pull request by lukechurch**
@marianamarasoiu would you mind reviewing this? 

It's adding a mapping table between a string and a random uuid, backed by Firestore. It's used for, amongst other things, maintaining a de-identification table for phone numbers.

### CODE REVIEW

File: id_infrastructure/firestore_uuid_table.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+class FirestoreUuidTable(object):
+    def __init__(self, table_name, crypto_token_path, uuid_prefix):
+        cred = credentials.Certificate(crypto_token_path)
+        firebase_admin.initialize_app(cred)
+        self._client = firestore.client()
+        self._table_name = table_name
+        self._uuid_prefix = uuid_prefix
+
+    def data_to_uuid_batch(self, list_of_data_requested):
+        # Steam the datastore to a local copy
```
Comment (marianamarasoiu) [inline id_infrastructure/firestore_uuid_table.py]: typo: s/steam/stream

File: id_infrastructure/firestore_uuid_table.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        # If it doesn't, create a new UUID, store it
+        # block until the store completes return the new UUID
+        uuid_doc_ref = self._client.document(u'tables/{}/mappings/{}'.format(self._table_name, data)).get()
+
+        exists = uuid_doc_ref.exists
+
+        print ("Ref: {}, exists: {}".format(uuid_doc_ref, exists))
+
+        if exists == False:
+            print ("No mapping found for: {}".format(data))
```
Comment (marianamarasoiu) [inline id_infrastructure/firestore_uuid_table.py]: This print statement doesn't seem to add much after the previous print statement. Would it make more sense to move it after the set() call and print the generated uuid as well?

File: id_infrastructure/firestore_uuid_table.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            )
+        else:
+            new_uuid = uuid_doc_ref.get("uuid")
+
+        return new_uuid
+    
+    def uuid_to_data(self, uuid_to_lookup):
+        # Search for the UUID
+        # return the data or fail
+        # Create a reference to the cities collection
```
Comment (marianamarasoiu) [inline id_infrastructure/firestore_uuid_table.py]: I don't think this comment should be here?

File: id_infrastructure/firestore_uuid_table.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+class FirestoreUuidTable(object):
+    def __init__(self, table_name, crypto_token_path, uuid_prefix):
+        cred = credentials.Certificate(crypto_token_path)
+        firebase_admin.initialize_app(cred)
+        self._client = firestore.client()
+        self._table_name = table_name
+        self._uuid_prefix = uuid_prefix
+
+    def data_to_uuid_batch(self, list_of_data_requested):
```
Comment (marianamarasoiu) [inline id_infrastructure/firestore_uuid_table.py]: After just reading this code I don't have a good sense as to what "data" is. The PR message adds a bit more context but once we merge that gets lost, so could you copy and paste the PR message at the top of this file?

File: id_infrastructure/firestore_uuid_table.py (hunk 1 above)
Comment (lukechurch) [reply id_infrastructure/firestore_uuid_table.py]: Fixed

File: id_infrastructure/firestore_uuid_table.py (hunk 2 above)
Comment (lukechurch) [reply id_infrastructure/firestore_uuid_table.py]: Done thanks

File: id_infrastructure/firestore_uuid_table.py (hunk 3 above)
Comment (lukechurch) [reply id_infrastructure/firestore_uuid_table.py]: cp + paste error :)

File: id_infrastructure/firestore_uuid_table.py (hunk 4 above)
Comment (lukechurch) [reply id_infrastructure/firestore_uuid_table.py]: Good idea - added thanks

**Comment by lukechurch** [comment]
Thanks @marianamarasoiu

**Pull request merged**